src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/AggregateRequestExecutor.java

```diff
@@ -81,32 +81,28 @@ public Response aggregate() throws Exception {
     final SortedMap<OSHDBTimestamp, ? extends Number> result;
     MapReducer<OSMEntitySnapshot> mapRed = null;
     mapRed = inputProcessor.processParameters();
+    var mapRedGeom = mapRed.map(OSMEntitySnapshot::getGeometry);
     switch (requestResource) {
       case COUNT:
         result = mapRed.aggregateByTimestamp().count();
         break;
-      case AREA:
-        result = mapRed.aggregateByTimestamp()
-            .sum((SerializableFunction<OSMEntitySnapshot, Number>) snapshot -> ExecutionUtils
-                .cacheInUserData(snapshot.getGeometry(), () -> Geo.areaOf(snapshot.getGeometry())));
-        break;
-      case LENGTH:
-        result = mapRed.aggregateByTimestamp()
-            .sum((SerializableFunction<OSMEntitySnapshot, Number>) snapshot -> ExecutionUtils
-                .cacheInUserData(snapshot.getGeometry(),
-                    () -> Geo.lengthOf(snapshot.getGeometry())));
-        break;
       case PERIMETER:
-        result = mapRed.aggregateByTimestamp()
-            .sum((SerializableFunction<OSMEntitySnapshot, Number>) snapshot -> {
-              if (snapshot.getGeometry() instanceof Polygonal) {
-                return ExecutionUtils.cacheInUserData(snapshot.getGeometry(),
-                    () -> Geo.lengthOf(snapshot.getGeometry().getBoundary()));
-              } else {
+        result = mapRedGeom.aggregateByTimestamp()
+            .sum(geom -> {
+              if (!(geom instanceof Polygonal)) {
                 return 0.0;
               }
+              return ExecutionUtils.cacheInUserData(geom, () -> Geo.lengthOf(geom.getBoundary()));
             });
         break;
+      case LENGTH:
+        result = mapRedGeom.aggregateByTimestamp()
+            .sum(geom -> ExecutionUtils.cacheInUserData(geom, () -> Geo.lengthOf(geom)));
+        break;
+      case AREA:
+        result = mapRedGeom.aggregateByTimestamp()
+            .sum(geom -> ExecutionUtils.cacheInUserData(geom, () -> Geo.areaOf(geom)));
+        break;
       default:
         throw new RuntimeException("Unsupported RequestResource type for this processing. "
             + "Only COUNT, LENGTH, PERIMETER, and AREA are permitted here");
```

Comment:
Is there a reason you re-sorted the cases?

Reply:
if I remember correctly, I did this to make the order the same for all of these very similar `switch (aggregationType)` statements. But apparently, I overlooked one which still has `LENGTH` and `PERIMETER` reversed.